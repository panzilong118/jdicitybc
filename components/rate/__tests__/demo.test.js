import * as React from "react";

import renderer from 'react-test-renderer';

import Basic from '../demo/Basic';

import Character from '../demo/Character';

import Clear from '../demo/Clear';

import Disabled from '../demo/Disabled';

import Half from '../demo/Half';

import Text from '../demo/Text';

describe('<Avatar />', () => {
    it('Basic 基本用法', () => {
        const demo = renderer.create(<Basic />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Character 其他字符', () => {
        const demo = renderer.create(<Character />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Clear star 清除', () => {
        const demo = renderer.create(<Clear />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Read only 只读', () => {
        const demo = renderer.create(<Disabled />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Half star 半星', () => {
        const demo = renderer.create(<Half />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Show copywriting 文案展现', () => {
        const demo = renderer.create(<Text />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });
});