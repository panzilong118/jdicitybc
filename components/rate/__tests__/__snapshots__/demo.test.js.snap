// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Basic 基本用法 1`] = `
<ul
  className="jc-rate"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
  tabIndex={0}
>
  <li
    className="jc-rate-star jc-rate-star-zero"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-zero"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-zero"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-zero"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-zero"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
</ul>
`;

exports[`<Avatar /> Character 其他字符 1`] = `
<div>
  <ul
    className="jc-rate"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-heart "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-heart "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-heart "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-heart "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-heart "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-heart "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-heart "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-heart "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-heart "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-heart "
          style={Object {}}
        />
      </div>
    </li>
  </ul>
  <br />
  <ul
    className="jc-rate"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "fontSize": 36,
      }
    }
    tabIndex={0}
  >
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        A
      </div>
      <div
        className="jc-rate-star-second"
      >
        A
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        A
      </div>
      <div
        className="jc-rate-star-second"
      >
        A
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        A
      </div>
      <div
        className="jc-rate-star-second"
      >
        A
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        A
      </div>
      <div
        className="jc-rate-star-second"
      >
        A
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        A
      </div>
      <div
        className="jc-rate-star-second"
      >
        A
      </div>
    </li>
  </ul>
  <br />
  <ul
    className="jc-rate"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        好
      </div>
      <div
        className="jc-rate-star-second"
      >
        好
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        好
      </div>
      <div
        className="jc-rate-star-second"
      >
        好
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        好
      </div>
      <div
        className="jc-rate-star-second"
      >
        好
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        好
      </div>
      <div
        className="jc-rate-star-second"
      >
        好
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        好
      </div>
      <div
        className="jc-rate-star-second"
      >
        好
      </div>
    </li>
  </ul>
</div>
`;

exports[`<Avatar /> Clear star 清除 1`] = `
<div>
  <ul
    className="jc-rate"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    <li
      className="jc-rate-star jc-rate-star-full"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-full"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-full"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
  </ul>
   allowClear: true
  <br />
  <ul
    className="jc-rate"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    <li
      className="jc-rate-star jc-rate-star-full"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-full"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-full"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
  </ul>
   allowClear: false
</div>
`;

exports[`<Avatar /> Half star 半星 1`] = `
<ul
  className="jc-rate"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
  tabIndex={0}
>
  <li
    className="jc-rate-star jc-rate-star-full"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-full"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-half jc-rate-star-active"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-zero"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-zero"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
</ul>
`;

exports[`<Avatar /> Read only 只读 1`] = `
<ul
  className="jc-rate jc-rate-disabled"
  onBlur={null}
  onFocus={null}
  onKeyDown={null}
  onMouseLeave={null}
  style={Object {}}
  tabIndex={-1}
>
  <li
    className="jc-rate-star jc-rate-star-full"
    onClick={null}
    onMouseMove={null}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-full"
    onClick={null}
    onMouseMove={null}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-zero"
    onClick={null}
    onMouseMove={null}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-zero"
    onClick={null}
    onMouseMove={null}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
  <li
    className="jc-rate-star jc-rate-star-zero"
    onClick={null}
    onMouseMove={null}
  >
    <div
      className="jc-rate-star-first"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
    <div
      className="jc-rate-star-second"
    >
      <i
        className="jcicon jcicon-star "
        style={Object {}}
      />
    </div>
  </li>
</ul>
`;

exports[`<Avatar /> Show copywriting 文案展现 1`] = `
<span>
  <ul
    className="jc-rate"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    <li
      className="jc-rate-star jc-rate-star-full"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-full"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-full"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
    <li
      className="jc-rate-star jc-rate-star-zero"
      onClick={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="jc-rate-star-first"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
      <div
        className="jc-rate-star-second"
      >
        <i
          className="jcicon jcicon-star "
          style={Object {}}
        />
      </div>
    </li>
  </ul>
  <span
    className="jc-rate-text"
  >
    3
     stars
  </span>
</span>
`;
